Preview Button: Remove the separator and border, and reduce the size of the icon.

shaunandrews opened the pull request:
There's a few visual things about the new device-preview button that have been bothering me. 

The first is the separator between the currently selected device, and the dropdown arrow. The line makes me thing that this button does two things; I'd expect different behavior when clicking the label and when clicking the icon. But both the label and the icon both trigger the same action. So, I don't see the need for the line at all. This PR removes it.

The border around button feels a little strange. Not all buttons in the toolbar have a border, in fact I don't see any other buttons in the top toolbar, or block toolbars that use a border. So, I removed the border and kind of like how it feels. What do you think?

The icon's current size feels too heavy. When compared to other dropdown in the editor and WordPress, the icon is larger than others. So, I reduces the size of the icon down to 16px and it feels nicer. What do you think?

Current:
<img width="417" alt="image" src="https://user-images.githubusercontent.com/191598/76098638-d1a25a80-5f97-11ea-8f0a-70c185128818.png">

This PR:
<img width="394" alt="image" src="https://user-images.githubusercontent.com/191598/76098388-593b9980-5f97-11ea-9efe-2595cef99064.png">

shaunandrews committed:
Remove the separator and border, and reduce the size of the icon.

This removes the separator and border from the preview toggle. It also reduces the size of the icon down to 16px. Along with this, some adjustments to the button padding were required.

karmatosed commented:
I really like this as gives some visual clarity. It also stops me wondering if the sizes are different on the drop-down and publish button.

mtias commented:
This looks good. I think we should always say "preview" there and not show the name of the current breakpoint.

mtias commented on packages/edit-post/src/components/preview-options/index.js:
```diff
@@ -22,6 +22,7 @@ import { external, check } from '@wordpress/icons';
 
 const downArrow = (
 	<SVG
+		className="editor-post-preview__button-icon"
```
This doesn't seem great

shaunandrews replied:
Teach me your ways. Whats wrong with it? We could totally do it without the className, but let me know why its bad.

mtias replied:
I mean this should be something that gets applied by default by using the right ingredients, without having to manually update the SVG.

shaunandrews committed:
Removing the className and the dynamic label. Now the button always says "Preview".

shaunandrews committed:
Move the svg sizing to the `header` component and adjusting the external icon's spacing and size.

shaunandrews committed:
Removed the "View" group label, put the external icon first, and updated the external label.

shaunandrews commented:
<img width="108" alt="image" src="https://user-images.githubusercontent.com/191598/76224994-41ac1d00-61f2-11ea-94b9-a140d45c1677.png">

I removed the dynamic label, replacing it with a static "Preview" label.

<img width="308" alt="image" src="https://user-images.githubusercontent.com/191598/76240806-411f8080-620a-11ea-995d-6a3186d7ab53.png">

I also removed the "View" label, repositioned the external icon, and updated the label from "Preview externally" to "Preview in new tab".

jasmussen commented on packages/edit-post/src/components/preview-options/style.scss:
```diff
@@ -53,11 +44,16 @@
+	justify-content: flex-start;
+
+	svg {
+		margin-right: 8px;
```
<img width="227" alt="Screenshot 2020-03-13 at 14 10 26" src="https://user-images.githubusercontent.com/1204802/76625216-0b92d400-6537-11ea-90bb-fa93bb4d9082.png">

This caused a few issues, so let's discuss this separately. In the mean time I removed it.

jasmussen committed:
Massage the pixels, update icon.

jasmussen committed:
Update snapshots.

jasmussen reviewed (approved):
I see a lot of positive reactions on this change, and good discussion. The PR was lovely, and nearly there. I took the liberty of massaging the pixels ever so slightly, and updating the External Link icon. Here's what it looks like now:

![i1](https://user-images.githubusercontent.com/1204802/76625501-abe8f880-6537-11ea-9f6c-8c85bfd5997c.gif)

This feels like a good evolution of the Preview dropdown. Let's try it!

mtias reviewed (approved):
Let's get this in, thanks for the iterations.

shaunandrews commented:
Thanks for taking this one to the finish line while I was AFK!
